src/org/traccar/api/BaseObjectResource.java

```diff
@@ -79,7 +80,10 @@ public Response add(T entity) throws SQLException {
             Context.getPermissionsManager().checkDeviceLimit(getUserId());
         } else if (baseClass.equals(Command.class)) {
             Context.getPermissionsManager().checkLimitCommands(getUserId());
-        } else if (entity instanceof ScheduledModel) {
+        } else if (entity instanceof GroupedModel && ((GroupedModel) entity).getGroupId() != 0) {
+            Context.getPermissionsManager().checkPermission(Group.class, getUserId(),
```

Comment:
It's a bit of a formatting preference, but I think if parameters don't fit it's better to new-line all parameters like this:
```java
Context.getPermissionsManager().checkPermission(
        Group.class, getUserId(), ((GroupedModel) entity).getGroupId());
```
This way you need method name on one line and all parameters on the other line. Easier to read. Also, some code editors handle it better.

Reply:
I like it in general with reserve. In case all parameters does not fit new line, may be better leave part on first line?

Reply:
In some rare case, but most of the time I would prefer two lines with just parameters and keep method on the first line.